File: ScriptService/Dto/Sense/TypeInfo.cs
using ScriptService.Extensions;

namespace ScriptService.Dto.Sense {

    /// <summary>
    /// info about a type
    /// </summary>
    public class TypeInfo : SenseEntry {

        /// <summary>
        /// type of element if this type is a collection
        /// </summary>
        public string ElementType { get; set; }

        /// <summary>
        /// methods available in type
        /// </summary>
        public MethodInfo[] Methods { get; set; }

        /// <summary>
        /// properties available in type
        /// </summary>
        public PropertyInfo[] Properties { get; set; }

        /// <summary>
        /// indexer methods
        /// </summary>
        public MethodInfo[] Indexer { get; set; }

        /// <inheritdoc />
        public override string ToString() {
            return Name.GetPlainTypeName();
        }
    }
}

File: nc-scripts/src/app/dto/sense/typeinfo.ts
import { MethodInfo } from './methodinfo';
import { PropertyInfo } from './propertyinfo';
import { SenseEntry } from './senseentry';

/**
 * info about a type
 */
export interface TypeInfo extends SenseEntry {

    /**
     * type of element if this type is a collection
     */
    elementType?: string,

    /**
     * methods available in type
     */
    methods: MethodInfo[],

    /**
     * properties available in type
     */
    properties: PropertyInfo[],

    /**
     * indexer methods 
     */
    indexer: MethodInfo[]
}